source/shared/support/_dynamic-search-bar.html.erb

```diff
@@ -0,0 +1,29 @@
+<script>
+  var filterInput = document.getElementById('search-table');
+  filterInput.addEventListener('keyup', filterResults);
+
+  function filterResults() {
+    var filterValue = filterInput.value.toUpperCase();
+    var rows = document.querySelectorAll('.result-row');
+
+    for(var i = 0; i < rows.length; i++) {
+      var text = rows[i].querySelectorAll('.govuk-table__cell')[0].innerHTML.toUpperCase();
+      rows[i].style.display = text.indexOf(filterValue) > -1 ? '' : 'none';
+    }
+
+    displayResultsMessage(rows);
+  }
+
+  function displayResultsMessage(results) {
+    var noResultsMessage = document.getElementById('no-results');
+    noResultsMessage.style.display = 'none';
+
+    var noResults = Array.from(results).every(
+      function(e) { return e.style.display == 'none'; }
+    );
+
+    if (noResults) {
+      noResultsMessage.style.display = 'block';
+    }
+  }
+</script>
```

Comment:
Given there is no ruby in this file, wonder if it makes sense to put this in a JS file rather than an ERB file and to include that?
Applies to the original admin implementation too, but can come back to that.

Reply:
Will do as an improvement